node/src/main/scala/coop/rchain/node/Main.scala

```diff
@@ -231,55 +229,72 @@ object Main {
   private def decryptKeyFromCon(
       encryptedPrivateKeyPath: Path
   )(implicit console: ConsoleIO[Task]): Task[PrivateKey] =
-    for {
-      password   <- ConsoleIO[Task].readPassword("Password for private key file: ")
-      privateKey <- Secp256k1.parsePemFile[Task](encryptedPrivateKeyPath, password)
-    } yield privateKey
+    Secp256k1.parsePemFile[Task](encryptedPrivateKeyPath, getValidatorPassword)
 
   private def generateKey(
-      algorithm: String,
-      privateKeyPath: Path,
-      publicKeyPath: Path
+      path: Path
   )(implicit console: ConsoleIO[Task]): Task[Unit] =
     for {
-      password       <- ConsoleIO[Task].readPassword("Password for generated private key file: ")
+      password       <- ConsoleIO[Task].readPassword("Enter password for keyfile: ")
       passwordRepeat <- ConsoleIO[Task].readPassword("Repeat password: ")
       _ <- if (password != passwordRepeat) {
-            ConsoleIO[Task].println("Passwords do not match. Try again.") >> generateKey(
-              algorithm,
-              privateKeyPath,
-              publicKeyPath
-            )
+            ConsoleIO[Task].println("Passwords do not match. Try again:") >> generateKey(path)
           } else if (password.isEmpty) {
-            ConsoleIO[Task].println("Password is empty. Try again.") >> generateKey(
-              algorithm,
-              privateKeyPath,
-              publicKeyPath
-            )
+            ConsoleIO[Task].println("Password is empty. Try again:") >> generateKey(path)
           } else {
             for {
-              sigAlgorithm <- SignaturesAlg(algorithm)
+              sigAlgorithm <- SignaturesAlg(Secp256k1.name)
```

Comment:
If algorithm is known in compile time it can be referenced directly, in this case `Secp256k1` instance.

Reply:
I just decided not to change much in case future we get back different signing algos. But I guess be you are right.